src/noit_metric_tag_search.c

```diff
@@ -37,10 +37,12 @@
 #include "noit_metric_tag_search.h"
 
 #include <stdio.h>
+#include <ck_pr.h>
 
 void
 noit_metric_tag_search_free(noit_metric_tag_search_ast_t *node) {
   if(node == NULL) return;
+  if(!ck_pr_dec_32_is_zero(&node->refcnt)) return;
```

Comment:
if it's not freed here, couldn't a zero here prevent it from eventually being freed?

Reply:
If there is a zero here, it would mean it has been freed and this would be a double free.